### ISSUE TITLE
Rule Request: Avoid optional Bools

### ISSUE THREAD

**Issue by ejmartin504** [realm/SwiftLint#2011]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

This is somewhat similar to #1885 -- optional Bools are similar to optional collections in that an "empty" Bool rarely conveys much information. 
I haven't found any existing discussions about this, but I have seen folks trying different solutions to dealing with optional Bools, all of which, to me, are not pretty. (see for example: Rule Request: Avoid optional collections)

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Would trigger:
```swift
let foo: Bool?
let bar: Bool? = false
let baz = Optional.some(false)
```

These would not trigger:
```swift
// Properties should provide default value
let foo: Bool = false

// functions should return Bool instead of optional Bool
func foo() -> Bool

// input params should default to true or false instead of being optional
func foo(bar: Bool = false)

// Optional Bool's can arise from collections
let jsonData: [String: Any]
let foo: Bool? = jsonData["foo"] as? Bool
```

3. Should the rule be configurable, if so what parameters should be configurable?

No

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

I'm curious what others think, but I would probably see this as an opt-in rule.

**Comment by marcelofabri** [comment]
Implemented in #2013
